x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetJobsStatsAction.java

```diff
@@ -75,7 +75,10 @@ public TransportGetJobsStatsAction(TransportService transportService, ActionFilt
     protected void doExecute(Task task, GetJobsStatsAction.Request request, ActionListener<GetJobsStatsAction.Response> finalListener) {
         logger.debug("Get stats for job [{}]", request.getJobId());
 
-        jobConfigProvider.expandJobsIds(request.getJobId(), request.allowNoJobs(), true, ActionListener.wrap(
+        ClusterState state = clusterService.state();
+        PersistentTasksCustomMetaData tasks = state.getMetaData().custom(PersistentTasksCustomMetaData.TYPE);
+        // If there are deleted configs, but the task is still around, we probably want to return the tasks in the stats call
```

Comment:
This logic isn't used in `TransportGetDatafeedsStats`.  Is there a good reason for that?

Reply:
`TransportGetDatafeedsStats` effectively requires the Datafeed config to exist because it returns stats related to the job, and there is now way to know the related job unless we have the config. 

So, I can attempt to change the _stats call for datafeeds (which probably means `null` timing stats for those only with tasks), or I can remove this from job stats. 

What do you think?

Reply:
I think often `TransportGetJobsStats` will return empty stats if a task exists but no config.  This is because for jobs with tasks it redirects to the node running the task to get the stats, but this node then returns empty stats if there is no `AutodetectCommunicator` (see `AutodetectProcessManager.getStatistics()`).  So for consistency we might as well report a datafeed with no stats in `TransportGetDatafeedsStats`.  At least then that will alert us to the orphaned task when looking at a support diag bundle.